### PR TITLE
[PATH] Drilling refactor part 1 - Generator and testing

### PR DESCRIPTION
**Pull request by sliptonic**
Refactoring the drilling operation.  This creates a new 'generator' for creating drilling gcode.  It introduces full testing for the generator but does not affect the current drill operation yet.

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [x]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

File: src/Mod/Path/Generators/drill_generator.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# ***************************************************************************
+# *   Copyright (c) 2021 sliptonic shopinthewoods@gmail.com                 *
```
Comment (luzpaz) [inline src/Mod/Path/Generators/drill_generator.py:3]: email should be enclosed by `<angle brackets>`

File: src/Mod/Path/Generators/drill_generator.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+# ***************************************************************************
+
+
+import PathScripts.PathLog as PathLog
+import Path
+import numpy
+
+__title__ = "Drilling Path Generator"
+__author__ = "sliptonic (Brad Collette)"
+__url__ = "http://www.freecadweb.org"
```
Comment (luzpaz) [inline src/Mod/Path/Generators/drill_generator.py:30]: `https://`

**Review by luzpaz — CHANGES_REQUESTED** [review]

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5154">pipeline status</a> for feature branch [PR_5154](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5154). Pipeline [406327306](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/406327306) was triggered at [4a7cdd5](https://github.com/FreeCAD/FreeCAD/pull/5154/commits/4a7cdd5095aa08e077901824ad73e7ad7c91faf4). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5154">pipeline status</a> for feature branch [PR_5154](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5154). Pipeline [408635921](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/408635921) was triggered at [73c01a4](https://github.com/FreeCAD/FreeCAD/pull/5154/commits/73c01a4196ebf4396971bf6a75a05bcd5f8141f4). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request merged**